search/includes/classes/class-search.php

```diff
@@ -1028,7 +1028,7 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 			add_site_option( $index_exists_option_name, $response );
 		}
 
-		if ( $is_cacheable ) {
+		if ( $is_cacheable && isset( $response_body_json ) && strlen( $response_body_json ) < 6 * MB_IN_BYTES ) {
```

Comment:
Do we have any data on what size it b0rks Memcached?

Reply:
yes we do. anything over 1MB compressed. The compression rate is usually 8-12 or so. We do a more complex check in alloptions-limit (see below) but I figured this will be cheap and easy. 

https://github.com/Automattic/vip-go-mu-plugins/blob/fdd99cf535e55eeda46d7a11a4cbb1603d8bc666/alloptions-limit.php#L60-L64

Do you have any particular concern about this approach?

Reply:
Sorry, I wasn't talking about the general overall limit, but if we have had an average of certain cached responses that did break Memcached, that's all. 🙂  Is it too expensive to calculate from the `gzdeflate()` length? Feels like that would be more accurate.